cognite/client/data_classes/postgres_gateway/users.py

```diff
@@ -0,0 +1,148 @@
+            else None,
+        )
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        output = super().dump(camel_case=camel_case)
+        if isinstance(self.credentials, SessionCredentials):
+            output["credentials"] = self.credentials.dump(camel_case=camel_case)
+
+        return output
+
+    def as_write(self) -> UserWrite:
+        return self
+
+
+class User(_UserCore):
+    """A user.
+
+    This is the read/response format of the user.
+
+    Args:
+        username (str): Username to authenticate the user on the DB.
+        created_time (int): The number of milliseconds since 00:00:00 Thursday, 1 January 1970, Coordinated Universal Time (UTC), minus leap seconds.
+        last_updated_time (int): The number of milliseconds since 00:00:00 Thursday, 1 January 1970, Coordinated Universal Time (UTC), minus leap seconds.
+        session_id (int): ID of the session tied to this user.
+
+    """
+
+    def __init__(self, username: str, created_time: int, last_updated_time: int, session_id: int) -> None:
+        self.username = username
+        self.created_time = created_time
+        self.last_updated_time = last_updated_time
+        self.session_id = session_id
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        return cls(
+            username=resource["username"],
+            created_time=resource["createdTime"],
+            last_updated_time=resource["lastUpdatedTime"],
+            session_id=resource["sessionId"],
+        )
+
+    def as_write(self) -> NoReturn:
+        raise TypeError(f"{type(self).__name__} cannot be converted to a write object")
+
+
+class UserUpdate(CogniteUpdate):
+    def __init__(
+        self,
+        username: str,
+    ) -> None:
+        super().__init__()
+        self.username = username
+
+    class _UpdateItemSessionCredentialsUpdate(CognitePrimitiveUpdate):
+        def set(self, value: SessionCredentials | None) -> UserUpdate:
+            return self._set(value.dump() if isinstance(value, SessionCredentials) else value)
+
+    @property
+    def credentials(self) -> UserUpdate._UpdateItemSessionCredentialsUpdate:
+        return self._UpdateItemSessionCredentialsUpdate(self, "credentials")
+
+    @classmethod
+    def _get_update_properties(cls, item: CogniteResource | None = None) -> list[PropertySpec]:
+        return [
+            PropertySpec("credentials", is_nullable=True),
+        ]
+
+    def dump(self, camel_case: Literal[True] = True) -> dict[str, Any]:
+        """Dump the instance into a json serializable Python data type.
+
+        Args:
+            camel_case (Literal[True]): No description.
+        Returns:
+            dict[str, Any]: A dictionary representation of the instance.
+        """
+        dumped: dict[str, Any] = {"update": self._update_object}
+        dumped["username"] = self.username
+        return dumped
```

Comment:
Just a suggestion. I'd at least vote to just create the dict directly with the two keys:
```suggestion
        return {"update": self._update_object, "username": self.username}
```